packages/agents/cartographer/poller/src/lib/operations/stableswap.ts

```diff
@@ -87,3 +87,70 @@ export const updateStableSwap = async () => {
     }
   }
 };
+
+export const updatePoolEvents = async () => {
+  const {
+    adapters: { subgraph, database },
+    logger,
+    domains,
+  } = getContext();
+  const { requestContext, methodContext } = createLoggingContext("updatePoolEvents");
+
+  const subgraphQueryMetaParams: Map<string, SubgraphQueryByTimestampMetaParams> = new Map();
+  const lastestBlockNumbers: Map<string, number> = await subgraph.getLatestBlockNumber(domains);
+
+  await Promise.all(
+    domains.map(async (domain) => {
+      let latestBlockNumber: number | undefined = undefined;
+      if (lastestBlockNumbers.has(domain)) {
+        latestBlockNumber = lastestBlockNumbers.get(domain)!;
+      }
+
+      if (!latestBlockNumber) {
+        logger.error("Error getting the latestBlockNumber for domain.", requestContext, methodContext, undefined, {
+          domain,
+          latestBlockNumber,
+          lastestBlockNumbers,
+        });
+        return;
+      }
+
+      // Retrieve the most recent stable swap exchange event we've saved for this domain.
+      const latestTimestamp = await database.getCheckPoint("stableswap_pool_events_timestamp_" + domain);
+      subgraphQueryMetaParams.set(domain, {
+        maxBlockNumber: latestBlockNumber,
+        fromTimestamp: latestTimestamp,
+        orderDirection: "asc",
+      });
+    }),
+  );
+
+  if (subgraphQueryMetaParams.size > 0) {
+    // Get stableswap pool events for all domains in the mapping.
+    const events = await subgraph.getStableSwapPoolEventsByDomainAndTimestamp(subgraphQueryMetaParams);
```

Comment:
Is there a need to paginate?
Can there be a lot of events in some cases, like cold start (db reset), or catchup after downtime ?

Thinking about it to make sure we don't hit graph or db query limits.